events/_AllArgs.h

```diff
@@ -0,0 +1,460 @@
+/* events
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_ALL_ARGS_H
+#define MBED_ALL_ARGS_H
+
+namespace events {
+
+/**
+ * AllArgs is a private utility class used by Task
+ *
+ * It allows binding arguments to a callable, function pointer or class.
+ */
+template<typename F, typename A1=void, typename A2=void, typename A3=void, typename A4=void, typename A5=void>
+struct AllArgs;
+
+template<typename B0>
+struct AllArgs<B0> {
```

Comment:
This can be split into two parts: 
* A tuple that is an heterogeneous container: Tuple<A, B, C, ...>
* A class that contains the static operations that copy the tuple content and executing it treating the first member as a callable and others as parameters.